sys/net/gnrc/netif/gnrc_netif_device_type.c

```diff
@@ -153,7 +153,15 @@ void gnrc_netif_ipv6_init_mtu(gnrc_netif_t *netif)
 #else /* IS_ACTIVE(CONFIG_GNRC_NETIF_NONSTANDARD_6LO_MTU) */
             netif->ipv6.mtu = IPV6_MIN_MTU;
 #endif /* IS_ACTIVE(CONFIG_GNRC_NETIF_NONSTANDARD_6LO_MTU) */
-            netif->sixlo.max_frag_size = MIN(SIXLOWPAN_FRAG_MAX_LEN, tmp);
+            if (tmp >= IPV6_MIN_MTU) {
```

Comment:
Shouldn't be better to compare to `netif->ipv6.mtu`?

Reply:
You mean

```suggestion
            if (tmp >= netif->ipv6.mtu) {
```

?

Reply:
Yes that way

Reply:
Needs to be compared with `netif->ipv6.mtu`, otherwise it will result on `max_frag_size` being set when using non-std MTU.

- `tmp >= netif->ipv6.mtu` will always be true when using the non std 6lo feature (tmp == mtu).
- When using a 802.15.4g device with a L2-PDU >= 1280 it will always be true of course.